app/code/Magento/Quote/Model/QuoteIdToMaskedQuoteId.php

```diff
@@ -19,43 +21,78 @@ class QuoteIdToMaskedQuoteId implements QuoteIdToMaskedQuoteIdInterface
      * @var QuoteIdMaskFactory
      */
     private $quoteIdMaskFactory;
-    /**
-     * @var CartRepositoryInterface
-     */
-    private $cartRepository;
 
     /**
      * @var QuoteIdMaskResource
      */
     private $quoteIdMaskResource;
+    /**
+     * @var CartInterfaceFactory
+     */
+    private $cartFactory;
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
 
     /**
      * @param QuoteIdMaskFactory $quoteIdMaskFactory
-     * @param CartRepositoryInterface $cartRepository
      * @param QuoteIdMaskResource $quoteIdMaskResource
+     * @param CartInterfaceFactory $cartInterfaceFactory
+     * @param StoreManagerInterface $storeManager
      */
     public function __construct(
         QuoteIdMaskFactory $quoteIdMaskFactory,
-        CartRepositoryInterface $cartRepository,
-        QuoteIdMaskResource $quoteIdMaskResource
+        QuoteIdMaskResource $quoteIdMaskResource,
+        CartInterfaceFactory $cartInterfaceFactory,
+        StoreManagerInterface $storeManager
     ) {
         $this->quoteIdMaskFactory = $quoteIdMaskFactory;
-        $this->cartRepository = $cartRepository;
         $this->quoteIdMaskResource = $quoteIdMaskResource;
+        $this->cartFactory = $cartInterfaceFactory;
+        $this->storeManager = $storeManager;
     }
 
     /**
      * @inheritDoc
      */
     public function execute(int $quoteId): string
     {
-        /* Check the quote exists to avoid database constraint issues */
-        $this->cartRepository->get($quoteId);
+        // Check the quote exists to avoid database constraint issues
+        $this->checkIfQuoteExists($quoteId);
 
         $quoteIdMask = $this->quoteIdMaskFactory->create();
         $this->quoteIdMaskResource->load($quoteIdMask, $quoteId, 'quote_id');
-        $maskedId = $quoteIdMask->getMaskedId() ?? '';
 
-        return $maskedId;
+        return $quoteIdMask->getMaskedId() ?? '';
+    }
+
+    /**
+     * This is a tiny implementation similar to the get method in \Magento\Quote\Api\CartRepositoryInterface
+     * that isn't loading all quote items, to ensure if quote exists
+     *
+     * @param int $quoteId
+     * @return void
+     * @throws NoSuchEntityException
+     */
+    private function checkIfQuoteExists(int $quoteId): void
+    {
+        $quote = $this->cartFactory->create();
+
+        $quote
+            ->setStoreId($this->storeManager->getStore()->getId())
+            ->loadByIdWithoutStore($quoteId);
```

Comment:
I found an issue - this method runs "_afterLoad" method, which then could cause triggering collect totals:
https://github.com/magento/magento2/blob/646cfbf/app/code/Magento/Quote/Model/Quote.php#L2464-L2468

Reply:
Fixed by using resource model